java/bulk-import/bulk-import-starter/src/main/java/sleeper/bulkimport/starter/executor/Executor.java

```diff
@@ -120,17 +129,26 @@ private void validateJob(BulkImportJob bulkImportJob) {
             failedChecks.add("The table name must be set to a non-null value.");
         } else if (!doesTableExist(bulkImportJob.getTableName())) {
             failedChecks.add("Table does not exist.");
+        } else if (!hasMinimumPartitions(stateStoreProvider, tablePropertiesProvider, bulkImportJob)) {
+            failedChecks.add("The minimum partition count was not reached");
         }
 
         if (null == bulkImportJob.getFiles() || bulkImportJob.getFiles().isEmpty()) {
             failedChecks.add("The input files must be set to a non-null and non-empty value.");
         }
 
+
         if (!failedChecks.isEmpty()) {
             String errorMessage = "The bulk import job failed validation with the following checks failing: \n"
                     + String.join("\n", failedChecks);
-
-            throw new IllegalArgumentException(errorMessage);
+            LOGGER.warn(errorMessage);
+            ingestJobStatusStore.jobRejected(taskId, bulkImportJob.toIngestJob(),
+                    validationTimeSupplier.get(), failedChecks);
+            return false;
+        } else {
+            ingestJobStatusStore.jobAccepted(taskId, bulkImportJob.toIngestJob(),
+                    validationTimeSupplier.get());
```

Comment:
The task ID doesn't seem to add anything here. If the validation event needs to have the same task ID as the other events in the run of the job, that will never happen because the task won't exist when the validation happens. I'm not sure how much of a problem this is for the model, but there's definitely no task at this point.

Reply:
The validation won't happen on the task for bulk import, because the Spark cluster doesn't necessarily exist yet, but it will happen on the task for ingest, because the job is only ever read in the ECS task.

Reply:
We could set the task ID that it's going to be on, but it needs to agree with the task ID that will be set on the Spark step.